### DISCUSSION TITLE
Query all channels

### DISCUSSION THREAD

**Discussion by kakashi01s** [GetStream/stream-chat-android#3688] [Q&A] [answered]
### Query all channels 

hello i was using query all channels by just providing filter of type 

        val request = QueryChannelsRequest(
            filter = Filters.or(
                Filters.eq("type", "messaging"),
            ),
            offset = 0,
            limit = 10,
            querySort = QuerySort.desc("last_message_at")
        ).apply {
            watch = true
            state = true
        }

        client.queryChannels(request).enqueue { result ->
            if (result.isSuccess) {
                channels.value = result.data()
                
            } else {
                // Handle result.error()
                
                // Toast.makeText(requireContext(), "Error Fetching Channels", Toast.LENGTH_SHORT).show()
            }
        }

but this is not working anymore and it asks for a filter of `in` 
but i need my users to join any desired channel 

Error:
 Status code: 403, with stream code: 70, description: QueryChannels failed with error: "2 channels match your query but cannot be returned because you don't have access to them. Did you forget to include {members: $in: ["XXXXXXXXXXXXXXXXXXXXXXXXXXXXX"]}?"

**Comment by filbabic** [comment, marked as answer]
Hey @kakashi01s!

We've had a change on the BE side for the default permissions and default channel queries for security/privacy reasons.

The way to go about this is to create your own custom channel type in the dashboard and when creating channels to configure them with that new type. Then in the dashboard settings, you can change the query channel and/or channel logic and users to be able to find the channels of this type without being in them.

Edit: The change happened in February 2021, so it's been around for a while!

Thanks!

**Reply by kakashi01s** [reply]
@filbabic Great ! It works , Thank You so much.
